$OpenBSD: patch-Source_WTF_wtf_Stopwatch_h,v 1.1 2015/03/27 18:38:45 ajacoutot Exp $

XXX push upstream; including <cmath> #undef isnan and provides std::isnan().
Stopwatch.h:47:58: error: 'isnan' was not declared in this scope

--- Source/WTF/wtf/Stopwatch.h.orig	Fri Mar 27 19:21:04 2015
+++ Source/WTF/wtf/Stopwatch.h	Fri Mar 27 19:21:25 2015
@@ -44,7 +44,7 @@ class Stopwatch : public RefCounted<Stopwatch> { (publ
 
     double elapsedTime();
 
-    bool isActive() const { return !isnan(m_lastStartTime); }
+    bool isActive() const { return !std::isnan(m_lastStartTime); }
 private:
     Stopwatch() { reset(); }
 
@@ -60,14 +60,14 @@ inline void Stopwatch::reset()
 
 inline void Stopwatch::start()
 {
-    ASSERT_WITH_MESSAGE(isnan(m_lastStartTime), "Tried to start the stopwatch, but it is already running.");
+    ASSERT_WITH_MESSAGE(std::isnan(m_lastStartTime), "Tried to start the stopwatch, but it is already running.");
 
     m_lastStartTime = monotonicallyIncreasingTime();
 }
 
 inline void Stopwatch::stop()
 {
-    ASSERT_WITH_MESSAGE(!isnan(m_lastStartTime), "Tried to stop the stopwatch, but it is not running.");
+    ASSERT_WITH_MESSAGE(!std::isnan(m_lastStartTime), "Tried to stop the stopwatch, but it is not running.");
 
     m_elapsedTime += monotonicallyIncreasingTime() - m_lastStartTime;
     m_lastStartTime = NAN;
@@ -75,7 +75,7 @@ inline void Stopwatch::stop()
 
 inline double Stopwatch::elapsedTime()
 {
-    bool shouldSuspend = !isnan(m_lastStartTime);
+    bool shouldSuspend = !std::isnan(m_lastStartTime);
     if (shouldSuspend)
         stop();
 
